examples/6_totp.py

```diff
@@ -0,0 +1,206 @@
+            # chat: "totp list"
+            ns = await self.__list(bot, team)
+            await bot.chat.send(channel, str(ns))
+            return
+
+        if not (len(msg) == 3 and msg[0] == self.MSG_PREFIX):
+            return
+
+        action, issuer = msg[1], msg[2]
+        if action == TotpMsg.PROVISION.value:
+            # chat: "totp provision <issuer>"
+            send_msg = "Error provisioning TOTP for {0}. If this issuer was previously provisioned, confirm reprovisioning with command `totp {1} {0}`.".format(
+                issuer, TotpMsg.REPROVISION.value
+            )
+            try:
+                uri = await self.__provision(bot, team, issuer, force=False)
+                send_msg = "TOTP provisioned for {}. provisioning_uri={}".format(
+                    issuer, uri
+                )
+            except Exception:
+                pass
+            finally:
+                await bot.chat.send(channel, send_msg)
+            return
+        if action == TotpMsg.REPROVISION.value:
+            # chat: "totp reprovision <issuer>"
+            send_msg = "Error reprovisioning TOTP for {}".format(issuer)
+            try:
+                uri = await self.__provision(bot, team, issuer, force=True)
+                send_msg = "TOTP reprovisioned for {}. provisioning_uri={}".format(
+                    issuer, uri
+                )
+            except Exception:
+                pass
+            finally:
+                await bot.chat.send(channel, send_msg)
+            return
+        if action == TotpMsg.NOW.value:
+            # chat: "totp now <issuer>"
+            send_msg = "Error getting current TOTP for {}".format(issuer)
+            code = await self.__now(bot, team, issuer)
+            if code:
+                send_msg = "Current TOTP for {}: {}".format(issuer, code)
+            await bot.chat.send(channel, send_msg)
+            return
+        if action == TotpMsg.URI.value:
+            # chat: "totp uri <issuer>"
+            send_msg = "Error getting provisioning_uri for {}".format(issuer)
+            uri = await self.__uri(bot, team, issuer)
+            if uri:
+                send_msg = uri
+            await bot.chat.send(channel, send_msg)
+            return
+        if action == TotpMsg.REMOVE.value:
+            # chat: "totp remove <issuer>"
+            send_msg = "No keys to remove for {}".format(issuer)
+            try:
+                await self.__remove(bot, team, issuer)
+                send_msg = "Removed TOTP keys for {}".format(issuer)
+            except Exception:
+                pass
+            finally:
+                await bot.chat.send(channel, send_msg)
+            return
+
+
+username = "yourbot"
+team = "yourtotpbotteam"
+
+bot = Bot(
+    username=username, paperkey=os.environ["KEYBASE_PAPERKEY"], handler=TotpHandler()
+)
+
+listen_options = {"filter-channels": [{"name": team, "members_type": "team"}]}
```

Comment:
is this the appropriate listen options?

Reply:
this is basically exactly what the options are on the CLI for listening to chats. you can see a bunch of examples if you do a `keybase chat api listen --help`

Reply:
ok to support implicit teams, i've removed the listen_options